app/jobs/shipit/github_sync_job.rb

```diff
@@ -17,13 +17,24 @@ def perform(params)
         @stack.transaction do
           shared_parent.try!(:detach_children!)
           new_commits.each do |gh_commit|
-            @stack.commits.create_from_github!(gh_commit)
+            append_commit(gh_commit)
           end
         end
       end
       CacheDeploySpecJob.perform_later(@stack)
     end
 
+    def append_commit(gh_commit)
+      appended_commit = @stack.commits.create_from_github!(gh_commit)
+      if appended_commit.revert?
+        impacted_commits = @stack.undeployed_commits.reverse.drop_while { |c| !appended_commit.revert_of?(c) }
+        impacted_commits.pop # appended_commit
```

Comment:
The pop is simply for this working state. Wehn we load the undeployed commits, the revert was just added on top. We don't want to lock the revert itself, so we pop it from the array.